From 1514c6a935e370d0ff4d05e23881398ad7ed1aab Mon Sep 17 00:00:00 2001
commit cecf5716d5c8e74aa5afaeec836db624d6f68879 
Author: Donald Sharp <sharpd@nvidia.com> 
Date:   Thu Aug 8 14:58:04 2024 -0400

commit cecf5716d5c8e74aa5afaeec836db624d6f68879
Author: Donald Sharp <sharpd@nvidia.com>
Date:   Thu Aug 8 14:58:04 2024 -0400

    lib: Don't print warning if not a daemon

    vtysh will print out the  upon
    every run of the program if the ulimit is set stupidly
    large.  Prevent this from being displayed for vtysh.

    Fixes: #16516
    Signed-off-by: Donald Sharp <sharpd@nvidia.com>
---
 lib/event.c  | 5 +++--
 lib/libfrr.c | 8 ++++++++
 lib/libfrr.h | 2 +-
 3 files changed, 12 insertions(+), 3 deletions(-)

diff --git a/lib/event.c b/lib/event.c
index fc46a11c0..be2637142 100644
--- a/lib/event.c
+++ b/lib/event.c
@@ -571,8 +571,9 @@ struct event_loop *event_master_create(const char *name)
 	}
 
 	if (rv->fd_limit > STUPIDLY_LARGE_FD_SIZE) {
-		zlog_warn("FD Limit set: %u is stupidly large.  Is this what you intended?  Consider using --limit-fds also limiting size to %u",
-			  rv->fd_limit, STUPIDLY_LARGE_FD_SIZE);
+		if (frr_is_daemon())
+			zlog_warn("FD Limit set: %u is stupidly large.  Is this what you intended?  Consider using --limit-fds also limiting size to %u",
+				  rv->fd_limit, STUPIDLY_LARGE_FD_SIZE);
 
 		rv->fd_limit = STUPIDLY_LARGE_FD_SIZE;
 	}
diff --git a/lib/libfrr.c b/lib/libfrr.c
index 2861ebe0b..5dd44c05f 100644
--- a/lib/libfrr.c
+++ b/lib/libfrr.c
@@ -1445,3 +1445,11 @@ void _libfrr_version(void)
 	write(1, banner, sizeof(banner) - 1);
 	_exit(0);
 }
+
+bool frr_is_daemon(void)
+{
+	if (di)
+		return true;
+
+	return false;
+}
\ No newline at end of file
diff --git a/lib/libfrr.h b/lib/libfrr.h
index ee436d9f8..d9025fe7e 100644
--- a/lib/libfrr.h
+++ b/lib/libfrr.h
@@ -188,7 +188,7 @@ extern const char *frr_get_progname(void);
 extern enum frr_cli_mode frr_get_cli_mode(void);
 extern uint32_t frr_get_fd_limit(void);
 extern bool frr_is_startup_fd(int fd);
-
+extern bool frr_is_daemon(void);
 /* call order of these hooks is as ordered here */
 DECLARE_HOOK(frr_early_init, (struct event_loop * tm), (tm));
 DECLARE_HOOK(frr_late_init, (struct event_loop * tm), (tm));
-- 
2.20.1

